scrapy/core/scheduler.py

```diff
@@ -256,6 +270,10 @@ def next_request(self) -> Optional[Request]:
         Increment the appropriate stats, such as: ``scheduler/dequeued``,
         ``scheduler/dequeued/disk``, ``scheduler/dequeued/memory``.
         """
+        delayed_request = self.dpqs.pop()
+        if delayed_request:
+            self.stats.inc_value('scheduler/dequeued/delayed/memory', spider=self.spider)
+            self._enqueue_request(delayed_request)
```

Comment:
To me, none of the described behaviors seem problematic. I think everything is OK as long as we make sure to explain this in detail in the documentation. And I think it is great that users can play with request priority to adjust the behavior to the liking.

Regarding the setting @VMRuiz proposes, I think we could have a setting for that effect, but I think we can go for something more flexible: implement a `DELAY_PRIORITY` setting, in line with [`DEPTH_PRIORITY`](https://docs.scrapy.org/en/latest/topics/settings.html#depth-priority), and make it 1 by default. Users who do not set priorities would get the delayed-requests-first behavior by default, even for FIFO queues. Although since LIFO is the default, maybe a default value of 0 for this setting may be OK, and less surprising.

Reply:
The issues brought up here are now documented, and `DELAY_PRIORITY_ADJUST` implemented, so I am hoping we can consider this resolved.